.github/workflows/build.yml

```diff
@@ -98,10 +98,12 @@ jobs:
     strategy:
       matrix:
         java: [ 11, 17 ]
-        junit-version: [ '5.7.2', '5.8.2' ]
         modular: [true, false]
         os: [ubuntu, macos, windows]
-    name: with Java ${{ matrix.java }}, JUnit ${{ matrix.junit-version }}, Modular ${{ matrix.modular }} on ${{ matrix.os }}
+        exclude:
+          - modular: false # this was run already in the previous stage, no need for a duplicated run
+            java: 11
```

Comment:
i will extract this into an own improvement, we can save 3 jobs by this :)

Reply:
i was talking about the exclude - we run those 3 jobs within our standard phase, we do not need to run them for java 11 and modular false again. 3 job less in the whole pipeline